src/main/java/net/bytzo/sessility/mixins/PlayerMixin.java

```diff
@@ -0,0 +1,132 @@
+							"play_noteblock", "tune_noteblock", "pot_flower", "trigger_trapped_chest", "enchant_item",
+							"play_record", "bell_ring")
+		.collect(Collectors.collectingAndThen(Collectors.toCollection(HashSet::new),Collections::unmodifiableSet));
+
+	/* optionally prints advancement debugging information */
+	private void advancement_debug(String state, String playername, String advancement) {
+		if (!Sessility.settings().properties().advancementDebugMessages) return;
+		LogManager.getLogger().info("{}: {} {}", playername, state, advancement);
+	}
+
+	/* use advancements as potential motility detectors */
+	private void processAdvancement(ServerPlayer player, String advancement, int count) {
+
+		// process advancement action list
+		if (Sessility.settings().properties().advancementActionDetection && advancementActionList.contains(advancement)) {
+			advancement_debug("[action]", player.getName().getString(), advancement);
+			player.resetLastActionTime();
+			return;
+		}
+
+		// process advancement interact list and wildcard filters
+		if (Sessility.settings().properties().advancementInteractDetection &&
+				 (advancementInteractList.contains(advancement) ||
+					advancement.startsWith("interact_with_") ||
+					advancement.startsWith("inspect_") ||
+					advancement.startsWith("open_")
+				 )
+			 ) {
+			advancement_debug("[interact]", player.getName().getString(), advancement);
+			player.resetLastActionTime();
+			return;
+		}
+
+		/* Unfortunately, advancements regarding movement are affected by non-player related actions.
+		 * If a reliable way to detect this is found, advancement.endsWith("_one_cm") might be usable. */
+	}
+
+	/* advancement hook for awardStat(ResourceLocation) */
+	@Inject(method = "awardStat(Lnet/minecraft/resources/ResourceLocation;)V", at = @At("HEAD"))
+	private void preAwardStat(ResourceLocation res, CallbackInfo callbackInfo) {
+		processAdvancement(((ServerPlayer)(Object)this), res.getPath(), 1);
+	}
+
+	/* advancement hook for awardStat(ResourceLocation, int) */
+	@Inject(method = "awardStat(Lnet/minecraft/resources/ResourceLocation;I)V", at = @At("HEAD"))
+	private void preAwardStat(ResourceLocation res, int value, CallbackInfo callbackInfo) {
+		processAdvancement(((ServerPlayer)(Object)this), res.getPath(), value);
+	}
+
+	/* end of advancement code */
+
+
+	@Unique
+	double last_xRot, last_yRot;
+
+	/* detect when a player rotates */
+	@Inject(method = "travel(Lnet/minecraft/world/phys/Vec3;)V", at = @At("HEAD"))
+	private void preTravel(Vec3 vec3, CallbackInfo callbackInfo) {
+		if (!Sessility.settings().properties().rotationTriggersMotility) return;
+
+		ServerPlayer player = (ServerPlayer)(Object)this;
+		if (player.getVehicle() != null || player.isPassenger()) return;	// doesn't work right with vehicles currently
+
+		double xRot = player.getXRot();
+		double yRot = player.getYRot();
+
+		// detect player rotation
+		if (last_xRot != xRot || last_yRot != yRot)
+			player.resetLastActionTime();
```

Comment:
Similar to the above, we might want to avoid calling `resetLastActionTime()` at could affect the vanilla server idle kick. Instead, you could use `setSessile(true)`.

Reply:
This and the other code was written from a perspective of detecting motility, and not sessility.  So in the case as above, it would actually be attempting to setSessile(false).